File: SwiftAlgorithm/BOJ/1700.swift
//
//  1700.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/03/19.
//

import Foundation

var input = readLine()!.components(separatedBy: " ").map { Int($0)! }
var arr = readLine()!.components(separatedBy: " ").map { Int($0)! }

let holeCount = input[0]
var multitap: [Int] = []
var result = 0

for (i, num) in arr.enumerated() {
    //이미 꽂혀 있음
    if multitap.contains(num) {
        continue
    }
    
    //구멍이 남음
    if multitap.count < holeCount {
        multitap.append(num)
        continue
    }
    
    //빼야 됨
    //안 쓰는 거 -> 가장 나중에 쓰는거
    var outIndex = -1
    let afterArr = arr[(i+1)...]
    //안 쓰는거
    for (k, multitapNumber) in multitap.enumerated() {
        if !afterArr.contains(multitapNumber) {
            outIndex = k
            break
        }
    }
    
    if outIndex == -1 {
        for multitapNumber in multitap {
            outIndex = max(outIndex, afterArr.firstIndex(of: multitapNumber)!)
        }
        outIndex = multitap.firstIndex(of: afterArr[outIndex])!
    }
    
    multitap.remove(at: outIndex)
    multitap.append(num)
    result += 1
}

print(result)

File: Python/BOJ/그리디/1700.py
//
//  1700.py
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/05.
//

import sys; readLine = sys.stdin.readline

# 멀티탭 구멍이 있으면 꽂음
# 멀티탭 구멍이 없으면 빼고 꽂아야 함
# 나중에 안 쓰는걸 뺌
# 나중에 안 쓰는게 없으면 가장 마지막에 쓰는걸 뽑음

n, k = map(int, readLine().split())
items = list(map(int, readLine().split()))

multitap = []
result = 0
for i in range(k):
  item = items[i]
  # 이미 꽂혀 있다면
  if item in multitap:
    continue

  # 구멍이 남아 있다면
  if len(multitap) < n:
    multitap.append(item)
    continue

  # 하나 빼야 함
  outIndex = -1
  afterUsed = items[i+1:]
  # 이후로 사용 안 하는거 뺌
  for j in range(len(multitap)):
    if not multitap[j] in afterUsed:
      outIndex = j
      break

  # 모두 다시 사용하면 가장 마지막에 쓰는걸 뺌
  if outIndex == -1:
    tmpIndex = 0
    for j in range(len(multitap)):
      usedIndex = afterUsed.index(multitap[j])
      if tmpIndex < usedIndex:
        tmpIndex = usedIndex
        outIndex = j
    
  multitap.remove(multitap[outIndex])
  multitap.append(item)
  result += 1

print(result)
